File: Patterns_inPython/pat22.py
n=int(input("enter the number"))
for i in range(0,2*n-1):
    for j in range(0,2*n-1):
        #top=i
        #left=j
        #right=(2*n-2)-j
        #down=(2*n-2)-i
        print(n-min(i,j,(2*n-2)-j,(2*n-2)-i),end="")
    print()

File: Patterns_in_CPP/pat22.cpp
#include<iostream>
using namespace std;
int main(){
	int i,n,j;
	cin>>n;
	for(i=0;i<2*n-1;i++){
		for(j=0;j<2*n-1;j++){
			int top=i;
            int left=j;
            int right=(2*n-2)-j;
            int down=(2*n-2)-i;
            cout<<n-min(min(top,down),min(left,right));
		}
		cout<<endl;
	}
}
